File: run/bak/go/output/Test.BenchCheck/Test_BenchCheck.go
package Test_BenchCheck

import (
	pkg_Bench "gopurs/output/Bench"
	pkg_Data_Semigroup "gopurs/output/Data.Semigroup"
	pkg_Data_Show "gopurs/output/Data.Show"
	pkg_Effect "gopurs/output/Effect"
	pkg_Effect_Console "gopurs/output/Effect.Console"
	gopurs_runtime "gopurs/output/gopurs_runtime"
	sync "sync"
)

var cache_act gopurs_runtime.Value
var once_act sync.Once
func Get_act() gopurs_runtime.Value {
	once_act.Do(func() {
		cache_act = gopurs_runtime.Apply2(gopurs_runtime.RecordGet(pkg_Effect.Get_bindEffect(), "bind"), pkg_Bench.Get_benchNow(), gopurs_runtime.Func(func(t1_0 gopurs_runtime.Value) gopurs_runtime.Value {
return gopurs_runtime.Apply2(gopurs_runtime.RecordGet(pkg_Effect.Get_bindEffect(), "bind"), pkg_Bench.Get_benchNow(), gopurs_runtime.Func(func(t2_1 gopurs_runtime.Value) gopurs_runtime.Value {
return gopurs_runtime.Apply(pkg_Effect_Console.Get_log(), gopurs_runtime.Apply2(gopurs_runtime.RecordGet(pkg_Data_Semigroup.Get_semigroupString(), "append"), gopurs_runtime.Str("Delta: "), gopurs_runtime.Apply(gopurs_runtime.RecordGet(pkg_Data_Show.Get_showNumber(), "show"), gopurs_runtime.Float((t2_1.FloatVal()) - (t1_0.FloatVal())))))
}))
}))
	})
	return cache_act
}

File: java_output/Test_BenchCheck.java
public class Test_BenchCheck {
    public static final Object FFI_STUB = new java.util.function.Function<Object, Object>() {
        public Object apply(Object arg) { return this; }
    };


public static final Object act = (new java.util.function.Supplier<Object>() { public Object get() { Object t1_0 = ((java.util.function.Supplier) (Object)(Bench.benchNow)).get(); Object t2_1 = ((java.util.function.Supplier) (Object)(Bench.benchNow)).get(); return ((java.util.function.Supplier) (Object)(((java.util.function.Function<Object, Object>) ((java.util.function.Function<Object, Object>) (arg) -> (java.util.function.Supplier<Object>) () -> { System.out.println(arg); return null; })).apply((((String) ("Delta: ")) + ((String) (((java.util.function.Function<Object, Object>) (Data_Show.showNumberImpl)).apply((((Double) (t2_1)) - ((Double) (t1_0)))))))))).get(); } });
}
